fix: update AV QC requirements

dheshanm opened the pull request:
From: Beau-Luke Colton

> ## Updates
> 
> 1. Please don’t flag the folder ‘Extraneous files’ 
> Along with anything inside these folders.
> FYI - This has been agreed upon by the Speech team to contain files that we want to keep, but are not audio files. These folders are only manually created by the Speech team, after review, to house these extraneous files. 
>  
> 2. Please don’t flag the folder ‘Additional interview files’ 
> Along with anything inside these folders.
> As above, this has been agreed upon by the Speech team to contain additional AV files that we want to keep, but we are not able to analyse at this time. These folders are only manually created by the Speech team, after review, to house these extra AV files. 
>  
> 3. Please don’t flag “recording.conf” files 
> The ProNET Speech team advised these files are not flagged as Out of SOP for them, please remove from ours. 
>  
> 4. Please don’t flag “chat.txt” files 
> Not necessary to flag.


This PR addresses these new requirements

dheshanm committed:
fix: update AV QC requirements

kcho commented on lochness/utils/path_checker.py:
```diff
@@ -235,6 +235,32 @@ def update_by_removing_unused_files(df: pd.DataFrame) -> None:
+        pd.DataFrame: The updated DataFrame.
+    """
+    for idx, row in df.iterrows():
+        if row["parent_dir"] == ("Extraneous files"):
```
@dheshanm if `Extraneous files` and `Additional interview files` have sub-directories, files under these sub-directories will be in the `df` framework and have `row["parent_dir"]` as the name of the sub-directory, not `Extraneous files` or `Additional interview files`. Maybe it is safer to use

```py
if "Extraneous files" in row["parent_dir"]:
```

dheshanm replied:
I now check if both "Extraneous files" and "Additional interview files" are in `row["parent_dir"]`:

```python
if "Extraneous files" in row["parent_dir"]:
    df.loc[idx, "file_check"] = True
elif "Additional interview files" in row["parent_dir"]:
    df.loc[idx, "file_check"] = True
```

dheshanm committed:
fix: Not excluding child directories within 'Extraneous files' and 'Additional interview files' from QC
Will not exclude all files and directories under 'Extraneous files' and'Additional interview files'

tashrifbillah commented on lochness/utils/path_checker.py:
```diff
@@ -235,6 +235,32 @@ def update_by_removing_unused_files(df: pd.DataFrame) -> None:
+    for idx, row in df.iterrows():
+        if "Extraneous files" in row["parent_dir"]:
+            df.loc[idx, "file_check"] = True
+        elif "Additional interview files" in row["parent_dir"]:
```
What is the guarantee that all subjects, sites will name them as `Extraneous files` , `Additional interview files`. Did you observe the same names across all subjects, sites collected so far?

tashrifbillah replied:
I see Beau-Luke's comment now. So resolving this conversation.

tashrifbillah commented:
(Removed his email address for privacy)

tashrifbillah commented on lochness/utils/path_checker.py:
```diff
@@ -235,6 +235,32 @@ def update_by_removing_unused_files(df: pd.DataFrame) -> None:
+    """
+    for idx, row in df.iterrows():
+        if "Extraneous files" in row["parent_dir"]:
+            df.loc[idx, "file_check"] = True
```
> Please don’t flag the folder 

I do not understand how this request is addressed by setting `file_check=True`. Also, did `file_check` column exists before?

dheshanm replied:
Yes! If `file_check=False`, that file is added to the list of files with SOP violations and is then emailed to the respective sites.

Here, I'm manually overriding `file_check` variable for the files we need to skip QC checks.

tashrifbillah reviewed (commented):
Tashrif asked clarifying questions.
